### PR TITLE
ci: change to use GitHub actions

### PR DESCRIPTION
**Pull request by FrankYang0529**
#### Which issue(s) this PR fixes:
https://github.com/longhorn/longhorn/issues/8039

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+name: build
+on:
+  push:
+    branches:
+    - master
+  pull_request:
+jobs:
+  build:
+    name: Build binaries
+    runs-on: ubuntu-latest
```
Comment (derekbit) [inline .github/workflows/build.yml:10]: Just curious, do we have other options?

Comment (FrankYang0529) [reply .github/workflows/build.yml:10]: Here, https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners/about-github-hosted-runners#standard-github-hosted-runners-for-public-repositories

**Comment by FrankYang0529** [comment]
Hi @derekbit, it looks like test failed because I added error check to avoid lint issue. Can we assert `err` is `nil` here?

https://github.com/longhorn/sparse-tools/blob/813ac9e807f839eabd9f3583f843fcfea1269f0d/sparse/test/client_test.go#L255-L256

**Comment by FrankYang0529** [comment]
> Hi @derekbit, it looks like test failed because I added error check to avoid lint issue. Can we assert `err` is `nil` here?
> 
> https://github.com/longhorn/sparse-tools/blob/813ac9e807f839eabd9f3583f843fcfea1269f0d/sparse/test/client_test.go#L255-L256

It looks like server returns `http: Server closed` when server is closed. I update the test case.

**Comment by FrankYang0529** [comment]
Hi @derekbit, I tried to remove error check, but there still has data race condition. May you take a look when you have time? Thank you.

https://github.com/longhorn/sparse-tools/actions/runs/8888699426/job/24405917283?pr=102

**Comment by shuo-wu** [comment]
I am fine with merging this then using another ticket to track the data race

**Pull request merged**